lib/uknetdev/netdev.c

```diff
@@ -139,40 +107,63 @@ static struct uk_netdev_einfo_overwrites *_alloc_einfo(struct uk_alloc *a,
 
 	/*
 	 * Parse IPv4 parameters
-	 * NOTE: `pick_arg` automatically returns NULL if an arguments is
-	 *       empty or does not exists.
+	 * NOTE: `uk_nextarg` automatically returns NULL if an arguments
+	 *       does not exists.
 	 */
-	_einfo->ip4.cidr     = pick_arg(&ipv4_conf[netdev_id], ':');
-	_einfo->ip4.gw       = pick_arg(&ipv4_conf[netdev_id], ':');
-	_einfo->ip4.dns0     = pick_arg(&ipv4_conf[netdev_id], ':');
-	_einfo->ip4.dns1     = pick_arg(&ipv4_conf[netdev_id], ':');
-	_einfo->ip4.hostname = pick_arg(&ipv4_conf[netdev_id], ':');
-	_einfo->ip4.domain   = pick_arg(&ipv4_conf[netdev_id], ':');
+	_einfo->ip4.cidr     = uk_nextarg(&ipv4_conf[netdev_id], ':');
+	_einfo->ip4.gw       = uk_nextarg(&ipv4_conf[netdev_id], ':');
+	_einfo->ip4.dns0     = uk_nextarg(&ipv4_conf[netdev_id], ':');
+	_einfo->ip4.dns1     = uk_nextarg(&ipv4_conf[netdev_id], ':');
+	_einfo->ip4.hostname = uk_nextarg(&ipv4_conf[netdev_id], ':');
+	_einfo->ip4.domain   = uk_nextarg(&ipv4_conf[netdev_id], ':');
 	/*
 	 * NOTE: We do not throw an error if additional arguments are handed
 	 *       over (after domain). This will keep this parsing code
 	 *       future-proof.
 	 */
 
-	if (_einfo->ip4.cidr)
-		uk_pr_debug("netdev%d: Overwrite einfo ip4.cidr: \"%s\"\n",
-			    netdev_id, _einfo->ip4.cidr);
-	if (_einfo->ip4.gw)
-		uk_pr_debug("netdev%d: Overwrite einfo ip4.gw: \"%s\"\n",
-			    netdev_id, _einfo->ip4.gw);
-	if (_einfo->ip4.dns0)
-		uk_pr_debug("netdev%d: Overwrite einfo ip4.dns0: \"%s\"\n",
-			    netdev_id, _einfo->ip4.dns0);
-	if (_einfo->ip4.dns1)
-		uk_pr_debug("netdev%d: Overwrite einfo ip4.dns1: \"%s\"\n",
-			    netdev_id, _einfo->ip4.dns1);
-	if (_einfo->ip4.hostname)
-		uk_pr_debug("netdev%d: Overwrite einfo ip4.host: \"%s\"\n",
-			    netdev_id, _einfo->ip4.hostname);
-	if (_einfo->ip4.domain)
-		uk_pr_debug("netdev%d: Overwrite einfo ip4.domain: \"%s\"\n",
-			    netdev_id, _einfo->ip4.domain);
-
+	if (_einfo->ip4.cidr) {
+		if (_einfo->ip4.cidr[0] == '\0')
+			_einfo->ip4.cidr = NULL;
+		else
```

Comment:
These all look like unnecessarily taken branches if PRINTD is not enabled :thinking:. Maybe we could surround this `else` with `#if`'s? Or, better yet, just place everything into one single `uk_pr_debug`? I see that `__uk_vsnprintf` can handle null strings. And then the `if`'s would just be
```C
if (_einfo->* && _einfo->ip4.*[0] == '\0')
```

Although it probably does not matter right? Perhaps DCE removes them anyways.

Reply:
Yes, my assumption is that the compiler will optimize these things out, even without DCE. So, I would not change it.

Reply:
I did now split it into two if-blocks. This may make it more readable because one if-nesting is removed (which is what you wanted to achieve). I assume the same here: the compiler will opt-out empty if statements.